src/openrct2/ReplayManager.cpp

```diff
@@ -379,6 +433,8 @@ namespace OpenRCT2
 
             gCurrentTicks = replayData->tickStart;
 
+            LoadAndCompareSnapshot(replayData->gameStateSnapshots);
```

Comment:
@duncanspumpkin You only ever load snapshots at predefined places. While the stream may hold more than two snapshots, you don't have a way of knowing where (in terms of ticks) they were taken. There might be such info embedded in the state itself, but I don't see how it could be consumed as it is now. That makes me believe the current implementation is not sufficient for holding arbitrary amounts of snapshots.

The call to LinkSnapshots has odd arguments, perhaps there is some magic happening inside there, but I can't tell.

Reply:
The first uint32_t of each snapshot is the tick number. If you wanted to use more you could take deserialise all of them using the existing deserialiser converting them into an array of `GameStateSnapshot_t`. You can then do what is needed.

Reply:
OK, that sounds like indeed extending the replays with more snapshots could be done without breaking the format.